### PR TITLE
Migrate L2 announcements and LB-IPAM to cilium

### PR DESCRIPTION
**Pull request by aaronmondal**
No need to use MetalLB since Cilium can do this as well.

- - -
This change is [Reviewable](https://reviewable.io/reviews/TraceMachina/nativelink/505)

### CODE REVIEW

**Review by aaronmondal — COMMENTED** [review]
+@MarcusSorealheis +@adam-singer +@allada 

> *[Reviewable](https://reviewable.io/reviews/TraceMachina/nativelink/505#-:-Nlyte5SBsIgfTqu0xnn:b-d6bogf)* status: 0 of 3 files reviewed, all discussions resolved (waiting on @adam-singer, @allada, and @MarcusSorealheis)

File: deployment-examples/kubernetes/00_infra.sh [hunk 1]
```diff
@@ -77,50 +80,41 @@ kubectl wait --for condition=Established crd/referencegrants.gateway.networking.
...
-  --version 0.13.12 \
-  --wait
+CILIUM_IP_CIDR=$(echo ${KIND_NET_CIDR} | sed "s@0.0/16@255.0/28@")
 
 cat <<EOF | kubectl apply -f -
 ---
-apiVersion: metallb.io/v1beta1
-kind: L2Advertisement
+apiVersion: cilium.io/v2alpha1
+kind: CiliumL2AnnouncementPolicy
```
Comment (MarcusSorealheis) [inline deployment-examples/kubernetes/00_infra.sh:104]: This is less stable not more stable, right? I think it's fine, but we need to ensure that neither our CI our future releases are impeded by an Alpha announcement policy.

File: deployment-examples/kubernetes/00_infra.sh [hunk 2]
```diff
@@ -77,50 +80,41 @@ kubectl wait --for condition=Established crd/referencegrants.gateway.networking.
...
+apiVersion: cilium.io/v2alpha1
+kind: CiliumLoadBalancerIPPool
 metadata:
   name: default-pool
-  namespace: metallb-system
 spec:
-  addresses:
-    - ${METALLB_IP_RANGE}
+  cidrs:
+    - cidr: ${CILIUM_IP_CIDR}
```
Comment (MarcusSorealheis) [inline deployment-examples/kubernetes/00_infra.sh:117]: Replacing MetalLb with Cilium is obviously an improvement.

**Review by MarcusSorealheis — APPROVED** [review]
LGTM. I'm totally fine with this one as it is only a deployment example.

**Review by aaronmondal — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/TraceMachina/nativelink/505)* status: 0 of 3 files reviewed, 2 unresolved discussions (waiting on @adam-singer, @allada, and @MarcusSorealheis)

___
*[`deployment-examples/kubernetes/00_infra.sh` line 104 at r1](https://reviewable.io/reviews/TraceMachina/nativelink/505#gh-1430695427:-Nlz2UT596Mw598DBhrB:bkujqp1) ([raw file](https://github.com/TraceMachina/nativelink/blob/ef8fd7fe750f2be5ad07abbd2a810c76f05236a6/deployment-examples/kubernetes/00_infra.sh#L104)):*
<details><summary><i>Previously, MarcusSorealheis (Marcus Eagan) wrote…</i></summary><blockquote>

This is less stable not more stable, right? I think it's fine, but we need to ensure that neither our CI our future releases are impeded by an Alpha announcement policy.
</blockquote></details>

From what I've tested it amounts to the same stability as in "just works", but the cilium variant is much faster to set up (e.g. in CI) since we don't need to fetch/deploy another helm chart.
___
*[`deployment-examples/kubernetes/00_infra.sh` line 117 at r1](https://reviewable.io/reviews/TraceMachina/nativelink/505#gh-1430695764:-Nlz2faZ7BYG4721wnlI:b-896fix) ([raw file](https://github.com/TraceMachina/nativelink/blob/ef8fd7fe750f2be5ad07abbd2a810c76f05236a6/deployment-examples/kubernetes/00_infra.sh#L117)):*
<details><summary><i>Previously, MarcusSorealheis (Marcus Eagan) wrote…</i></summary><blockquote>

Replacing MetalLb with Cilium is obviously an improvement.
</blockquote></details>

Done.

**Pull request merged**